pkg/trace/agent/agent.go

```diff
@@ -349,6 +355,23 @@ func newChunksArray(chunks []*pb.TraceChunk) []*pb.TraceChunk {
 
 var _ api.StatsProcessor = (*Agent)(nil)
 
+// filterSpans removes all spans for which the provided FilterSpan function returns true
+func (a *Agent) filterSpans(p *api.Payload) {
+	if a.FilterSpan == nil {
+		return
+	}
+	for i := 0; i < len(p.Chunks()); i++ {
+		chunk := p.Chunk(i)
+		filteredSpans := []*pb.Span{}
+		for _, span := range chunk.Spans {
+			if !a.FilterSpan(span) {
+				filteredSpans = append(filteredSpans, span)
+			}
+		}
+		chunk.Spans = filteredSpans
```

Comment:
What is the desired outcome when all spans are filtered from a chunk? Do we delete the chunk or do we keep the empty chunk?

Reply:
Hey! I'm taking over this PR but I'm not entirely sure which one would be better. My gut is that we should delete the chunk but I'm not entirely sure what that would entail. Is there any reason not to delete the chunk?

Reply:
This function will leave the chunk empty, but then the empty chunk will be discarded by existing logic [here](https://github.com/DataDog/datadog-agent/pull/11687/files#diff-c2dc8d00a060f89f4cd810fa2496eb721f7da1a6f538c4b2ec5c5f7a894a76e7R213)